addons/pos_viva_wallet/controllers/main.py

```diff
@@ -0,0 +1,29 @@
+# coding: utf-8
+import logging
+import json
+from odoo import http, _
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class PosVivaWalletController(http.Controller):
+    @http.route('/pos_viva_wallet/notification', type='http', auth='none', csrf=False)
+    def notification(self, company_id, token):
+        _logger.info('notification received from Viva Wallet')
+
+        payment_method_sudo = request.env['pos.payment.method'].sudo().search([('use_payment_terminal', '=', 'viva_wallet'), ('company_id.id', '=', company_id), ('viva_wallet_webhook_verification_key', '=', token)], limit=1)
+
+        if payment_method_sudo:
+            if request.httprequest.data:
```

Comment:
if payment_method_sudo and request.httprequest.data:

Reply:
yes but no
because if there not data i need to respond with viva_wallet_webhook_verification_key
and if there are not payment_method_sudo i must respond with viva_wallet_webhook_verification_key